cmake/hip.cmake

```diff
@@ -0,0 +1,43 @@
+if(NOT WITH_AMD_GPU)
+    return()
+endif()
+
+include_directories("/opt/rocm/include")
+include_directories("/opt/rocm/hipblas/include")
+include_directories("/opt/rocm/hiprand/include")
+include_directories("/opt/rocm/rocrand/include")
+include_directories("/opt/rocm/rccl/include")
+include_directories("/opt/rocm/thrust")
+
+list(APPEND EXTERNAL_LIBS "-L/opt/rocm/lib/ -lhip_hcc")
+
+set(HIP_HCC_FLAGS "${HIP_HCC_FLAGS} -fPIC -DPADDLE_WITH_HIP -std=c++14" )
```

Comment:
Should we use `c++14` by default? Paddle is currently using `c++11`

Reply:
@sabreshao , could you check the `c++14` option here? This is not required by HIP as well.

Reply:
@sunway513 @reyoung c++14 is to enable "function return type deduction in lambda", which may be needed in following PR.